cmd/api/config.go

```diff
@@ -55,12 +55,17 @@ type config struct {
 			KeepFiles int  `default:"5"` // number of files to keep
 		}
 	}
-	TelemetryPublisher struct {
-		Enabled            bool   `default:"false"`
-		MetricsHubURL      string `default:""`
-		MetricsHubAPIKey   string `default:""`
-		PublishingInterval string `default:"10s"`
-	}
+	TelemetryPublisher TelemetryPublisherConfig
```

Comment:
I extracted the struct to be fully-defined. It will become clear why in `main` comments.